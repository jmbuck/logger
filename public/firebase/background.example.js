// Initialize Firebase
var config = {
    apiKey: "EXAMPLE",
    authDomain: "EXAMPLE",
    databaseURL: "EXAMPLE",
    projectId: "EXAMPLE",
    storageBucket: "EXAMPLE",
    messagingSenderId: "EXAMPLE"
};
firebase.initializeApp(config);

/**
 * initApp handles setting up the Firebase context and registering
 * callbacks for the auth status.
 *
 * The core initialization is in firebase.App - this is the glue class
 * which stores configuration. We provide an app name here to allow
 * distinguishing multiple app instances.
 *
 * This method also registers a listener with firebase.auth().onAuthStateChanged.
 * This listener is called when the user is signed in or out, and that
 * is where we update the UI.
 *
 * When signed in, we also authenticate to the Firebase Realtime Database.
 */
function initApp() {
    // Listen for auth state changes.
    firebase.auth().onAuthStateChanged(function(user) {
        console.log('User state change detected from the Background script of the Chrome Extension:', user);
    });
}

window.onload = function() {
    initApp();
};