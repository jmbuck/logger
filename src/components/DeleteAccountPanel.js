import React, { Component } from 'react'
import '../css/common.css'
import exitIcon from "../img/x.svg";
import { auth } from "../database/Auth";
import {deleteFirebaseAccount} from "../logger-firebase";

class DeleteAccountPanel extends Component {

    deleteAccount(uid) {
        deleteFirebaseAccount(uid)
    }

    render() {
        return (<div className="panel">
                <div className="panel-container">
                    <div className="panel-left-nav" />
                    <div className="panel-center-content">
                        <div className="Title">
                            Are you sure? You will lose all associated data and there is no way to retrieve this again.
                        </div>
                        <div className="Two-Grid">
                            <div onClick={() => this.deleteAccount(auth.currentUser.uid)}>
                                Yes, I'm sure
                            </div>
                            <div onClick={ () => this.props.history.push("/preferences") }>
                                No, take me back
                            </div>
                        </div>
                    </div>

                    <div className="panel-right-nav">
                        <div className="exit-nav" onClick={ () => this.props.history.push("/data") }>
                            <img src={exitIcon} />
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}

export default DeleteAccountPanel