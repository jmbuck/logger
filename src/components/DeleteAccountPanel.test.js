import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import DeleteAccountPanel from "./DeleteAccountPanel";
import firebase from "firebase";

Enzyme.configure({ adapter: new Adapter() });

describe("DeleteAccountPanel", () => {
    beforeAll(() => {
        firebase.auth = jest.fn().mockReturnValue({
            onAuthStateChanged: true
        });
    });

    it("should render component", () => {
        const component = Enzyme.shallow(<DeleteAccountPanel />);

        expect(component).toMatchSnapshot();
    });
});

