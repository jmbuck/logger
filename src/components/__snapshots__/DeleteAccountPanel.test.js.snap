// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteAccountPanel should render component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DeleteAccountPanel />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="Panel-Container"
>
        <div
                className="Delete-Account-Panel"
        >
                <div
                        className="Title"
                >
                        Are you sure? You will lose all associated data and there is no way to retrieve this again.
                </div>
                <div
                        className="Two-Grid"
                >
                        <div>
                                Yes, I'm sure
                        </div>
                        <div
                                onClick={[Function]}
                        >
                                No, take me back
                        </div>
                </div>
        </div>
</div>,
      "className": "Panel",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="Delete-Account-Panel"
>
          <div
                    className="Title"
          >
                    Are you sure? You will lose all associated data and there is no way to retrieve this again.
          </div>
          <div
                    className="Two-Grid"
          >
                    <div>
                              Yes, I'm sure
                    </div>
                    <div
                              onClick={[Function]}
                    >
                              No, take me back
                    </div>
          </div>
</div>,
        "className": "Panel-Container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="Title"
>
              Are you sure? You will lose all associated data and there is no way to retrieve this again.
</div>,
            <div
              className="Two-Grid"
>
              <div>
                            Yes, I'm sure
              </div>
              <div
                            onClick={[Function]}
              >
                            No, take me back
              </div>
</div>,
          ],
          "className": "Delete-Account-Panel",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Are you sure? You will lose all associated data and there is no way to retrieve this again.",
              "className": "Title",
            },
            "ref": null,
            "rendered": "Are you sure? You will lose all associated data and there is no way to retrieve this again.",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div>
                  Yes, I'm sure
</div>,
                <div
                  onClick={[Function]}
>
                  No, take me back
</div>,
              ],
              "className": "Two-Grid",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Yes, I'm sure",
                },
                "ref": null,
                "rendered": "Yes, I'm sure",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "No, take me back",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "No, take me back",
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="Panel-Container"
>
          <div
                    className="Delete-Account-Panel"
          >
                    <div
                              className="Title"
                    >
                              Are you sure? You will lose all associated data and there is no way to retrieve this again.
                    </div>
                    <div
                              className="Two-Grid"
                    >
                              <div>
                                        Yes, I'm sure
                              </div>
                              <div
                                        onClick={[Function]}
                              >
                                        No, take me back
                              </div>
                    </div>
          </div>
</div>,
        "className": "Panel",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="Delete-Account-Panel"
>
            <div
                        className="Title"
            >
                        Are you sure? You will lose all associated data and there is no way to retrieve this again.
            </div>
            <div
                        className="Two-Grid"
            >
                        <div>
                                    Yes, I'm sure
                        </div>
                        <div
                                    onClick={[Function]}
                        >
                                    No, take me back
                        </div>
            </div>
</div>,
          "className": "Panel-Container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="Title"
>
                Are you sure? You will lose all associated data and there is no way to retrieve this again.
</div>,
              <div
                className="Two-Grid"
>
                <div>
                                Yes, I'm sure
                </div>
                <div
                                onClick={[Function]}
                >
                                No, take me back
                </div>
</div>,
            ],
            "className": "Delete-Account-Panel",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Are you sure? You will lose all associated data and there is no way to retrieve this again.",
                "className": "Title",
              },
              "ref": null,
              "rendered": "Are you sure? You will lose all associated data and there is no way to retrieve this again.",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div>
                    Yes, I'm sure
</div>,
                  <div
                    onClick={[Function]}
>
                    No, take me back
</div>,
                ],
                "className": "Two-Grid",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Yes, I'm sure",
                  },
                  "ref": null,
                  "rendered": "Yes, I'm sure",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "No, take me back",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "No, take me back",
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
